### PR TITLE
[Process Agent] Disable console logging when running `check --json`

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

This PR fixes json output with the `check --json` command. You can now pipe to `jq`!

### Describe how to test/QA your changes

Run `process-agent check process --json` and ensure that there is no console output besides the json. You can also quickly verify that the output is proper json by running `process-agent check process --json | jq` and making sure there are no errors.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]

File: cmd/process-agent/app/check.go [hunk 1]
```diff
@@ -137,7 +142,9 @@ func runCheck(cfg *config.AgentConfig, ch checks.Check) error {
 
 	time.Sleep(1 * time.Second)
 
-	printResultsBanner(ch.Name())
+	if !checkOutputJSON {
+		printResultsBanner(ch.Name())
+	}
```
Comment (xornivore) [inline cmd/process-agent/app/check.go:147]: Sanity check - on line 151 below - when we return an error - this does end up in stderr? 🙇

**Review by xornivore — APPROVED** [review]
🙇

**Review by robertjli — APPROVED** [review]
LGTM! One question about the doc, and also I'm curious what you think about a future improvement—
What if we changed the `checkOutputJSON` variable into `outputFormat` and `verbosity` variables, so that we can explicitly request meta-output with some sort of `--verbose` flag?

File: releasenotes/notes/process-agent-fix-json-output-bc58ab39de316bc4.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+fixes:
+  - `process-agent check --json` now outputs valid json.
```
Comment (robertjli) [inline releasenotes/notes/process-agent-fix-json-output-bc58ab39de316bc4.yaml:3]: The `--json` flag is newly introduced in this release, so this note might be a bit misleading/confusing given that it was never "broken".

@maycmlee curious what you think?

Comment (just-chillin) [reply releasenotes/notes/process-agent-fix-json-output-bc58ab39de316bc4.yaml:3]: That's a great point. I was personally going back and forth on whether or not to include this for that exact reason. While the `--json` output is newly added, in fact json output has been broken for many releases, and so I figured it was at least worth mentioning.

I am also curious about what @maycmlee has to say about this.

File: cmd/process-agent/app/check.go (hunk 1 above)
Comment (just-chillin) [reply cmd/process-agent/app/check.go:147]: Yep! We're only suppressing the console output for seelog here, so everything else will get printed.

I tested erroring unconditionally to ensure that cobra prints to `stderr` when returning an error, and it did, so we are good on that.

**Comment by just-chillin** [comment]
> LGTM! One question about the doc, and also I'm curious what you think about a future improvement— What if we changed the `checkOutputJSON` variable into `outputFormat` and `verbosity` variables, so that we can explicitly request meta-output with some sort of `--verbose` flag?

I agree that this solution is better, but I would be interested in talking to Agent Core first since we want to try to keep parity with the core agent. I think if we were to implement this in the future, we would have to make `--json` an alias for `--output json`, to avoid any compatibility problems in the future

File: releasenotes/notes/process-agent-fix-json-output-bc58ab39de316bc4.yaml (hunk 2 above)
Comment (maycmlee) [reply releasenotes/notes/process-agent-fix-json-output-bc58ab39de316bc4.yaml:3]: Hi! What about something like, "Adds new `--json` flag. Running `process-agent check --json` outputs valid json."

Comment (just-chillin) [reply releasenotes/notes/process-agent-fix-json-output-bc58ab39de316bc4.yaml:3]: I think that will work. While the `--json` flag was added in https://github.com/DataDog/datadog-agent/pull/11480/files#diff-dd84c63abd7d921936a48fc4f088b6dd01d9cd529e54f9961de44720c3094c80R3, assuming it all gets concatenated together I don't see a problem with it.

**Pull request merged**